src/ShoppinPal/Vend/Api/V2/Products.php

```diff
@@ -68,4 +68,33 @@ public function get($productId)
         return new Product($result['data'], Product::UNKNOWN_PROPERTY_IGNORE, true);
 
     }
+
+    /**
+     * Returns the true, that update image.
+     *
+     * @param string $productId ID of the product.
+     * @param string $imagePath of the image.
+     *
+     * @return true
+     *
+     * @throws EntityNotFoundException If the product is not found.
+     */
+    public function imageUpload($productId, $imagePath)
+    {
+        $request = $this->getAuthenticatedRequestForUri('api/2.0/products/' . urlencode($productId) . '/actions/image_upload');
+
+        $request->setMethod(CurlHttpRequest::METHOD_POST);
+
+        $request->addHeader('Content-Type: multipart/form-data');
+
+        $request->setPayload(
+            json_encode(array("image"=>fopen($imagePath, 'r'))),
```

Comment:
You need to send a multipart/form data request, that doesn't expect a json. So json_encoding makes no sense.

Reply:
fopen creates a file handle resource. You can't send that to Vend. You need the actual image data.

Reply:
You need to create a valid multipart/form request string here if you use raw. Raw means it send the provided string as the request body without any modifications.